go/tasks/plugins/array/k8s/config.go

```diff
@@ -31,14 +37,80 @@ var (
 		},
 	}
 
-	configSection = config.MustRegisterSection(configSectionKey, defaultConfig)
+	configSection = pluginsConfig.MustRegisterSubSection(configSectionKey, defaultConfig)
 )
 
+type ResourceConfig struct {
+	PrimaryLabel string `json:"primaryLabel" pflag:",PrimaryLabel of a given service cluster"`
+	Limit        int    `json:"limit" pflag:",Resource quota (in the number of outstanding requests) for the cluster"`
+}
+
+type ClusterConfig struct {
```

Comment:
can we put some comments here, explaining why we have a cluster config and why we need to do this? This seems a little complicated IMO

Reply:
Sure.